### ISSUE TITLE
About the calculation of mrcnn_mask_loss

### ISSUE THREAD

**Issue by keineahnung2345** [matterport/Mask_RCNN#978]
In `mrcnn_mask_loss_graph`, `y_true` and `y_pred`'s shape are both `[N, height, width]`. When using `K.binary_crossentropy` to calculate mrcnn_mask_loss, it will return a tensor with shape `[N, height]`, meaning that it treat every row in the mask as a sample, so there are `N*height` samples and each sample's length is `width`. 
But the loss should be calculated on `N` samples with each sample of size `height*width`.  So I think `y_true` and `y_pred` should be reshaped into `[N, height*width]` before calculating the loss. Any ideas?

**Comment by waleedka** [comment]
The binary cross-entropy is applied element-wise. So the loss is of shape `[N, height, width]`. Then K.mean() collapses it to one value.

**Comment by keineahnung2345** [comment]
Sorry, you are right, I originally expected that the behavior of tf.keras.losses.binary_crossentropy and K.binary_crossentropy will be the same. So I experimented with tf.keras.losses.binary_crossentropy, and it gave me output of shape [N, height]. When I later experimented with K.binary_crossentropy, it gave me output of shape [N, height, width], just as you told.
